File: fixtures/node_output/put_basic_auth_json_data.js
var request = require('request');

var dataString = '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}';

var options = {
    url: 'http://localhost:5984/test/_security',
    method: 'PUT',
    body: dataString,
    auth: {
        'user': 'admin',
        'pass': '123'
    }
};

function callback(error, response, body) {
    if (!error && response.statusCode == 200) {
        console.log(body);
    }
}

request(options, callback);

File: fixtures/scrapy_output/put_basic_auth_json_data.py
from scrapy import Request, Spider
from w3lib.http import basic_auth_header


class MySpider(Spider):
    name = 'myspider'

    def start_requests(self):
        auth = basic_auth_header(
            'admin',
            '123',
        )
        headers = {
            'Authorization': auth,
        }
        body = '{"admins":{"names":[], "roles":[]}, "readers":{"names":["joe"],"roles":[]}}'
        yield Request(
            'http://localhost:5984/test/_security',
            callback=self.parse,
            method='PUT',
            headers=headers,
            body=body,
        )

    def parse(self, response):
        pass
